src/test/scala/it/pagopa/interop/agreementprocess/AgreementUpdateSpec.scala

```diff
@@ -0,0 +1,125 @@
+package it.pagopa.interop.agreementprocess
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import it.pagopa.interop.agreementmanagement.client.model.UpdateAgreementSeed
+import it.pagopa.interop.agreementprocess.model.AgreementUpdatePayload
+import org.scalatest.matchers.should.Matchers._
+import org.scalatest.wordspec.AnyWordSpecLike
+
+import java.util.UUID
+
+class AgreementUpdateSpec extends AnyWordSpecLike with SpecHelper with ScalatestRouteTest {
+
+  import agreementApiMarshaller._
+
+  "Agreement Update" should {
+    "succeed if agreement is in draft" in {
+
+      val eServiceId   = UUID.randomUUID()
+      val descriptorId = UUID.randomUUID()
+      val consumerId   = UUID.randomUUID()
+
+      val payload = AgreementUpdatePayload("consumer-notes")
+
+      val agreement =
+        SpecData.draftAgreement.copy(
+          eserviceId = eServiceId,
+          descriptorId = descriptorId,
+          consumerId = consumerId,
+          producerId = requesterOrgId,
+          consumerNotes = Some("old-consumer-notes")
+        )
+
+      val updatedAgreement =
+        agreement.copy(consumerNotes = Some(payload.consumerNotes))
+
+      val seed = UpdateAgreementSeed(
+        state = agreement.state,
+        certifiedAttributes = agreement.certifiedAttributes,
+        declaredAttributes = agreement.declaredAttributes,
+        verifiedAttributes = agreement.verifiedAttributes,
+        suspendedByConsumer = agreement.suspendedByConsumer,
+        suspendedByProducer = agreement.suspendedByProducer,
+        suspendedByPlatform = agreement.suspendedByPlatform,
+        consumerNotes = Some(payload.consumerNotes),
+        stamps = agreement.stamps
+      )
+
+      mockAgreementRetrieve(agreement)
+      mockAgreementUpdate(agreement.id, seed, updatedAgreement)
+
+      Get() ~> service.updateAgreementById(agreement.id.toString, payload) ~> check {
+        status shouldEqual StatusCodes.OK
+      }
+    }
+  }
+
+  "Agreement Update" should {
+    "fail if agreement is not in draft" in {
```

Comment:
Same here, how can I tell the agreement is not in draft state?

Reply:
You tell that this agreement is not in draft state from this line (using SpecData)
`val agreement = SpecData.activeAgreement.copy(...)`